pkg/devfile/generator/generators.go

```diff
@@ -91,6 +98,70 @@ func GetContainers(devfileObj parser.DevfileObj, options common.DevfileOptions)
 	return containers, nil
 }
 
+// GetInitContainers gets the init container for every preStart devfile event
+func GetInitContainers(devfileObj parser.DevfileObj) ([]corev1.Container, error) {
+	containers, err := GetContainers(devfileObj, common.DevfileOptions{})
+	if err != nil {
+		return nil, err
+	}
+	preStartEvents := devfileObj.Data.GetEvents().PreStart
+	var initContainers []corev1.Container
+	if len(preStartEvents) > 0 {
+		var eventCommands []string
+		commands, err := devfileObj.Data.GetCommands(common.DevfileOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		commandsMap := getCommandsMap(commands)
+
+		for _, event := range preStartEvents {
+			eventSubCommands := getCommandsFromEvent(commandsMap, strings.ToLower(event))
+			eventCommands = append(eventCommands, eventSubCommands...)
+		}
+
+		for i, commandName := range eventCommands {
+			if command, ok := commandsMap[commandName]; ok {
```

Comment:
if you think about it, we no longer have to do this check because of devfile/api validation where we check if the events are valid but i guess no harm in having it here too but removing it would remove the usage of commandsMap which is less code 🤔

Reply:
It's not a validation check, it's finding the particular event(command) using the event(command)'s name. If not saving it into a map, we need to iterate through the command list evenytime

Reply:
oops